org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/XtextProjectHelper.java

```diff
@@ -37,7 +40,7 @@ public static boolean hasNature(IProject project) {
 	 * @since 2.4
 	 */
 	public static boolean hasBuilder(IProject project) {
-		if (project.isAccessible()) {
+		if (project != null && project.isAccessible()) {
```

Comment:
I don't like this here. We should bail out eagerly since it's most likely a caller problem.

Reply:
Updated to use `Preconditions.checkNotNull(project)`